services/components/add-component-handler.js

```diff
@@ -20,21 +21,66 @@ function getParentListElement(el, path) {
 }
 
 /**
+ * determine if a component is available on the current site, based on logic
+ * @param {string} slug e.g. "grubstreet"
+ * @param {string} logic e.g. (di, press, thecut) or (not:di, not:press, not:thecut) or a mixture of both
+ * @returns {boolean}
+ */
+function availableOnCurrentSite(slug, logic) {
+  var tokens = logic.split(',').map((str) => str.trim()), // trim any extra whitespace
+    // list of site slugs to include
+    include = _.reject(tokens, (token) => _.includes(token, 'not:')),
```

Comment:
could you call these "sitesToInclude" "sitesToExclude"? Self-documenting variable names!